bundles/binding/org.openhab.binding.expire/src/main/java/org/openhab/binding/expire/internal/ExpireBinding.java

```diff
@@ -0,0 +1,156 @@
+     * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+     */
+    public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+        this.bundleContext = bundleContext;
+
+        setProperlyConfigured(true);
+    }
+
+    /**
+     * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+     *
+     * @param configuration Updated configuration properties
+     */
+    public void modified(final Map<String, Object> configuration) {
+        // update the internal configuration accordingly
+    }
+
+    /**
+     * Called by the SCR to deactivate the component when either the configuration is removed or
+     * mandatory references are no longer satisfied or the component has simply been stopped.
+     *
+     * @param reason Reason code for the deactivation:<br>
+     *            <ul>
+     *            <li>0 – Unspecified
+     *            <li>1 – The component was disabled
+     *            <li>2 – A reference became unsatisfied
+     *            <li>3 – A configuration was changed
+     *            <li>4 – A configuration was deleted
+     *            <li>5 – The component was disposed
+     *            <li>6 – The bundle was stopped
+     *            </ul>
+     */
+    public void deactivate(final int reason) {
+        this.bundleContext = null;
+        // deallocate resources here that are no longer needed and
+        // should be reset when activating this binding again
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected long getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected String getName() {
+        return "Expire Refresh Service";
+    }
+
+    protected static class ExpireConfig {
+        protected long expireAfterMs;
+
+        protected long nextExpireTs;
+    }
+
+    protected static Map<String, Long> nextExpireTsMap = new HashMap<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void execute() {
+        for (ExpireBindingProvider provider : providers) {
+            for (String itemName : provider.getItemNames()) {
+                Long nextExpireTs = nextExpireTsMap.get(itemName);
+                if (nextExpireTs == null) {
+                    // set initial expireTs
+                    long expireAfterMs = provider.getExpiresAfterMs(itemName);
+                    nextExpireTsMap.put(itemName, System.currentTimeMillis() + expireAfterMs);
+                } else if (nextExpireTs <= System.currentTimeMillis()) {
+                    // value expired
+                    State newState = provider.getExpiredState(itemName);
+                    logger.info("Item {} was not updated for {} - setting state to {}", itemName,
+                            provider.getExpiresAfterAsText(itemName), newState);
```

Comment:
I'm not sure here. It triggers an ItemStateEvent that logs that the item state changed to UNDEF (or whatever configured). ItemStateEvent logs as Info. So IMO it would be good to see, why it was set to UNDEF.

Example log:
```

2016-11-26 20:41:17.098 [INFO ] [smarthome.event.ItemStateEvent:43   ] - Temperature_Test2 updated to 64
2016-11-26 20:41:17.098 [INFO ] [s.event.ItemStateChangedEvent :43   ] - Temperature_Test2 changed from UNDEF to 64
2016-11-26 20:41:25.696 [INFO ] [.expire.internal.ExpireBinding:137  ] - Item Temperature_Test1 was not updated for 10s  - setting state to undefined
2016-11-26 20:41:25.697 [INFO ] [smarthome.event.ItemStateEvent:43   ] - Temperature_Test1 updated to UNDEF
2016-11-26 20:41:25.698 [INFO ] [s.event.ItemStateChangedEvent :43   ] - Temperature_Test1 changed from 7 to UNDEF
```